shbin.py

```diff
@@ -60,6 +61,21 @@ def get_repo_and_user():
     return gh.get_repo(os.environ["SHBIN_REPO"]), gh.get_user().login
 
 
+def get_repo(full_name_repo):
+    gh = Github(os.environ["SHBIN_GITHUB_TOKEN"])
+    return gh.get_repo(full_name_repo)
+
+
+def get_path(url):
+    pattern = r"https://github.com/.*?/(?:tree|blob)/(.*)"
+    result = re.findall(pattern, url)
```

Comment:
do not use `findall` for a single potential match. also it would be a little better to compile the pattern at module level 

```
PATTERN = re.compile(r"https://github.com/.*?/(?:tree|blob)/(.*)")

def get_path(url):
    try:
         return re.match(pattern, url).group(1)
    except ...     
```